File: _REPO/APRESS/distributed-.net-programming-in-vb-.net/Chapter09/BinaryFormatter/Sender/SenderMain.vb
Imports System.Messaging
Imports CustomerLibrary

Module SenderMain

   Sub Main()
      'Create the queue instance
      Dim mq As New MessageQueue(".\private$\MyPrivateQ")

      Dim msg As New Message()
      msg.Label = "A Customer object"
      msg.Formatter = New BinaryMessageFormatter()

      Do
         'Construct Customer and send to queue
         msg.Body = New Customer("Homer", "hsimpson@atomic.com", "5555")
         mq.Send(msg)
      Loop Until Console.ReadLine() = "q"

   End Sub

End Module

File: _REPO/APRESS/distributed-.net-programming-in-csharp/Chapter09/MoreXMLFormatter/Sender/SenderMain.cs
using System;
using System.Messaging;
using CustomerLibrary;

namespace Sender
{	
   class SenderMain
   {	
      static void Main(string[] args)
      {        
         // Create the queue instance
         MessageQueue mq = new MessageQueue(@".\private$\MyPrivateQ");
         
         Message msg = new Message();
         msg.Label = "Customer Message";

         do
         {
            // Construct Customer and send to queue
            msg.Body = new Customer("Homer", "hsimpson@atomic.com", "5555");     
            mq.Send(msg);
         } while(Console.ReadLine() != "q");
      }
   }
}
